spec/factories/shift.rb

```diff
@@ -6,8 +6,11 @@
 
     trait :full do
       after(:build) do |shift|
+        day_begin = shift.day.beginning_of_day
+        day_end = shift.day.end_of_day
+
         ShiftEvent::ACTIONS.keys.zip(
-          Array.new(4) { Faker::Time.between(shift.day, shift.day) }.sort
+          Array.new(4) { Faker::Time.between(day_begin, day_end) }.sort
```

Comment:
We `zip` with `ACTIONS` to get one event of each action-type, so the `sort` ensures that the sign-in occurs before the sign-out.